File: cprograms/binarysearch.c
// Linear Search
#include <stdio.h>
#define SIZE 5
void linearSearch(int arr[SIZE], int);
int main()
{
    int arr[SIZE];
    int target;
    // read the array
    for (int i = 0; i < SIZE; i++)
    {
        printf("\nEnter the element %d : ", i + 1);
        scanf("%d", &arr[i]);
    }
    // take the element you want to search
    printf("\nEnter the element to search: ");
    scanf("%d", &target);
    // call the function
    linearSearch(arr, target);
    return 0;
}
void linearSearch(int arr[SIZE], int target)
{
    for (int i = 0; i < SIZE; i++)
    {
        if (target != arr[i])
        {
            continue;
        }
        else
        {
            printf("\nElement found!");
            return;
        }
    }
    printf("Element not found");
}

File: Recursion/src/BinarySearch.java
public class BinarySearch {
    public static void main(String[] args) {
        int[] nums = {23, 34, 55, 78, 99, 102, 226, 245};
        System.out.println(search(nums, 102, 0, nums.length - 1));
    }
    static int search(int[] arr, int target, int s, int e){
        if (s > e){
            return -1;
        }
        int mid = s + (e - s) / 2;
        if (arr[mid] == target){
            return mid;
        }
        if (arr[mid] > target){
            return search(arr, target, s, mid - 1);
        }
        else{
            return search(arr, target, mid + 1, e);
        }
    }
}
